client/webpack.hot.config.js

```diff
@@ -1,25 +1,28 @@
 // Run like this:
 // cd client && node server.js
 
+const webpack = require('webpack');
 const path = require('path');
 const config = require('./webpack.common.config');
-const webpack = require('webpack');
 
 // We're using the bootstrap-sass loader.
 // See: https://github.com/shakacode/bootstrap-sass-loader
-config.entry.push('webpack-dev-server/client?http://localhost:3000',
+config.entry.app.unshift(
+  'webpack-dev-server/client?http://localhost:3000',
   'webpack/hot/dev-server',
   './scripts/webpack_only',
 
   // custom bootstrap
-  'bootstrap-sass!./bootstrap-sass.config.js');
+  'bootstrap-sass!./bootstrap-sass.config.js'
+);
+
 config.output = {
 
   // this file is served directly by webpack
-  filename: 'express-bundle.js',
+  filename: '[name].js',
   path: __dirname,
 };
-config.plugins = [new webpack.HotModuleReplacementPlugin()];
+config.plugins.unshift(new webpack.HotModuleReplacementPlugin());
```

Comment:
In the past I had a problem concerned with order of plugins or loaders. I don't remember details, but since then I apply plugins & loaders in order from my templatet.